### PR TITLE
Remove Pylint unused-import warning suppressions

### PR DESCRIPTION
**Pull request by IRDonch**
### Summary

This makes all static analyzers happy (rather than just pylint) and ensures
we don't miss any unexpected unused-import issues in the affected files.

### How to test

N/A

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/util/image.py [hunk 1]
```diff
@@ -18,7 +17,7 @@ class _IMAGE_BACKENDS(Enum):
 _IMAGE_BACKEND = None
 _image_loading_errors = (FileNotFoundError, )
 try:
-    import cv2
+    importlib.import_module('cv2')
     _IMAGE_BACKEND = _IMAGE_BACKENDS.cv2
 except ImportError:
     import PIL
```
Comment (zhiltsov-max) [inline datumaro/util/image.py:23]: Probably, `PIL` import also deserves a change.

**Review by zhiltsov-max — COMMENTED** [review]
I'm not sure I like the way it is implemented. It looks like a fight with linting tools.

File: datumaro/util/image.py (hunk 1 above)
Comment (IRDonch) [reply datumaro/util/image.py:23]: The `PIL` import is not unused, so I can't change it.

**Comment by IRDonch** [comment]
> I'm not sure I like the way it is implemented. It looks like a fight with linting tools.

Why? I think it's a fairly logical convention that imported names are only for use within the imported module, and if you want them to be a part of the module's interface, you should re-export them explicitly.

At least in the case of `datumaro/util/__init__.py`, I consider my version easier to understand than the original (for humans as well as linters). It definitely took me some headscratching to understand the purpose of the original import.

File: datumaro/util/__init__.py [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 #
 # SPDX-License-Identifier: MIT
 
 from contextlib import ExitStack
-from distutils.util import strtobool as str_to_bool # pylint: disable=unused-import
+import distutils.util
 from functools import partial, wraps
 from itertools import islice
 from typing import Iterable, Tuple
 import attr
```
Comment (zhiltsov-max) [inline datumaro/util/__init__.py:11]:
```suggestion
from functools import partial, wraps
from itertools import islice
from typing import Iterable, Tuple
import distutils.util

import attr
```

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**